tools/ldb_cmd.cc

```diff
@@ -1812,8 +1816,19 @@ void InternalDumpCommand::DoCommand() {
 
     if (!count_only_ && !count_delim_) {
       std::string key = ikey.DebugString(is_key_hex_);
-      std::string value = Slice(key_version.value).ToString(is_value_hex_);
-      std::cout << key << " => " << value << "\n";
+      Slice value = Slice(key_version.value);
+      if (!decode_blob_index_ || value_type != kTypeBlobIndex) {
+        std::cout << key << " => " << value.ToString(is_value_hex_) << "\n";
```

Comment:
Seems to me that the `idump` output is written using `fprintf`. It would probably make sense to stick with it because `fprintf` and the C++ streams both buffer output independently, so if we start mixing them, their output might get reordered/interleaved in unexpected ways.

Reply:
I see. I have updated them to all use fprintf.